### PR TITLE
Capstone Project

### PR DESCRIPTION
**Pull request by renercrows**
I created a webpage calling a Pokemon API and returning the data, I used Redux and did Jest testing on it.

### Click the logo to get to the home page, if not you won't be able to see pokemons.

Video:
https://www.loom.com/share/6dabf9cf2b074f9192e45ea4d64557b0

Update: the go-back button is the logo.

Update2: I eliminated the Api file where I was using some API data, and also eliminated the API data from the SinglePokemon file, at last, I Stored the data of the API I used on the SingleCard File.

### The video is a little outdated, I made some changes since the video was done, see Github files for the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Miroverse Inc.
+
```
Comment (Davidosky007) [inline README.md:43]: * Kindly ensure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: public/index.html [hunk 2]
```diff
@@ -1,43 +0,0 @@
...
-<html lang="en">
-  <head>
-    <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <meta name="theme-color" content="#000000" />
-    <meta
-      name="description"
-      content="Web site created using create-react-app"
-    />
```
Comment (Davidosky007) [inline public/index.html]: * Kindly add back your public folder and index.html file as without this your react app wont be able to run, i can see you have deleted them 🙏

File: src/components/Api.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      data.results.forEach(async (el) => {
+        const res = await fetch(el.url);
+        const json = await res.json();
+        const pokemon = {
+          id: json.id,
+          name: json.name,
+          avatar: json.sprites.front_shiny,
+          stats: json.stats[0].base_stat,
+        };
+        setPokemons((pokemons) => [...pokemons, pokemon]);
```
Comment (Davidosky007) [inline src/components/Api.js]: * Kindly ensure that all data retrieved from the API should be stored in the Redux store this is a[ requirement ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20data%20retrieved%20from%20the%20API%20should%20be%20stored%20in%20the%20Redux%20store)

File: src/components/DisplayPokemons.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import SingleCard from './SingleCard';
+
+function DisplayPokemons() {
+  const pokemons = useSelector((state) => state.preview);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!pokemons.length) dispatch(fetchApi());
+  });
+  // console.log(pokemons);
```
Comment (Davidosky007) [inline src/components/DisplayPokemons.js]: * Kindly remove all commented code from your codebase as it makes your code bulky and looks unprofessional

File: src/components/SingleCard.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+function SingleCard({ pokemon }) {
+  const [img, setImg] = useState(pokeball);
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await axios.get(`https://pokeapi.co/api/v2/pokemon/${pokemon.id}`);
+      setImg(response.data.sprites.other['official-artwork'].front_default);
+    }
+    fetchData();
```
Comment (Davidosky007) [inline src/components/SingleCard.js]: * Kindly ensure that all data retrieved from the API should be stored in the Redux store [Checkout the technical requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20data%20retrieved%20from%20the%20API%20should%20be%20stored%20in%20the%20Redux%20store)

File: src/components/SinglePokemon.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+function SinglePokemons() {
+  const [img, setImg] = useState(pokeball100);
+
+  useEffect(() => {
+    async function fetchData() {
+      const random = Math.floor(Math.random() * 150) + 1;
+      const response = await axios.get(`https://pokeapi.co/api/v2/pokemon/${random}`);
+      setImg(response.data.sprites.versions.animated.front_default);
+    }
```
Comment (Davidosky007) [inline src/components/SinglePokemon.js]: * Kindly ensure that all data retrieved from the API should be stored in the Redux store [Checkout the technical requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20data%20retrieved%20from%20the%20API%20should%20be%20stored%20in%20the%20Redux%20store)

File: src/index.css [hunk 7]
```diff
@@ -1,27 +1,338 @@
...
+}
+
+.main-continer {
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (Davidosky007) [inline src/index.css:17]: * Currently when i click on the details page i can't find the back button kindly add this as it improves the user experience of your application [Check details page requirement here ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=When%20the%20user%20clicks%20on%20the%20%22Back%22%20button%20(%3C)%2C%20the%20user%20navigates%20to%20the%20home%20page.)
image

File: src/pages/PokemonData.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  const [abilities, setAbilities] = useState([]);
+  const [moves, setMoves] = useState([]);
+  const [types, setTypes] = useState([]);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await axios.get(
+        `https://pokeapi.co/api/v2/pokemon/${params.url}`,
+      );
```
Comment (Davidosky007) [inline src/pages/PokemonData.js:23]: * Kindly ensure that all data retrieved from the API should be stored in the Redux store [Checkout the technical requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20data%20retrieved%20from%20the%20API%20should%20be%20stored%20in%20the%20Redux%20store)

File: src/utils/FetchCards.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+const getPokemons = async () => {
+  const url = 'https://pokeapi.co/api/v2/pokemon/';
+  const answer = await fetch(url);
+  const data = await answer.json();
+  return data;
+};
```
Comment (Davidosky007) [inline src/utils/FetchCards.js]: * Kindly ensure that all data retrieved from the API should be stored in the Redux store [Checkout the technical requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20data%20retrieved%20from%20the%20API%20should%20be%20stored%20in%20the%20Redux%20store)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @renercrows 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Good video presentation 👍 
* All tests passed ✔️ 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/SingleCard.js (hunk 5 above)
Comment (smunozmo) [reply src/components/SingleCard.js]: - [ ] Your app is looking great, congrats on it! 🎉 However, as the previous reviewer said, the data fetched from the API should be stored in the Redux store:

Screen Shot 2022-09-08 at 11 57 49 PM

File: src/components/SinglePokemon.js (hunk 6 above)
Comment (smunozmo) [reply src/components/SinglePokemon.js]: - [ ] You are almost there! Let's try to store the retrieved data from the API in the Redux store as the previous reviewer said.

File: src/components/DisplayPokemons.js (hunk 4 above)
Comment (smunozmo) [reply src/components/DisplayPokemons.js]: - [ ] Your app is working properly, excellent job! I would suggest cleaning up any unused code like this comment to keep it professional as this is a cool project for your portfolio.

File: src/components/Api.js (hunk 3 above)
Comment (smunozmo) [reply src/components/Api.js]: - [ ] Nice job, you are almost ready! It would be great if you addressed this change suggested by the previous reviewer.

**Review by smunozmo — CHANGES_REQUESTED** [review]
Hi @renercrows, 👋  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights 👏 

- You have done an excellent job on this project, congrats on it! 👏 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._




Cheers and Happy coding! 🤓 

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mateo951 — APPROVED** [review]
### Hi @renercrows, 

Well done implementing all the changes required by the previous reviewer 🔥
In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it 🚀 

### Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**